`__stderrp` and friends not defined on OpenBSD

When trying to compile Nim on OpenBSD, you hit errors like the following:

```
/home/euant/.cache/nim/koch_d/stdlib_system.nim.c:1656:11: error: use of undeclared identifier '__stderrp'
        rawWrite(__stderrp, "out of memory\012");
                 ^
/home/euant/.cache/nim/koch_d/stdlib_system.nim.c:5751:11: error: use of undeclared identifier '__stderrp'
        rawWrite(__stderrp, msg);
                 ^
/home/euant/.cache/nim/koch_d/stdlib_system.nim.c:5945:12: error: use of undeclared identifier '__stderrp'
                rawWrite(__stderrp, "[GC] cannot register thread local variable; too many thread loc"
                         ^
/home/euant/.cache/nim/koch_d/stdlib_system.nim.c:6099:12: error: use of undeclared identifier '__stderrp'
                rawWrite(__stderrp, "[GC] cannot register global variable; too many global variables");
                         ^
/home/euant/.cache/nim/koch_d/stdlib_system.nim.c:7149:11: error: use of undeclared identifier '__stderrp'
        rawWrite(__stderrp, nimToCStringConv(errormsg));
                 ^
/home/euant/.cache/nim/koch_d/stdlib_system.nim.c:7151:11: error: use of undeclared identifier '__stderrp'
        rawWrite(__stderrp, "\012");
                 ^
6 errors generated.
Error: execution of an external program failed: 'clang -c  -w  -I/home/euant/Nim/lib -I/home/euant/Nim -o /home/euant/.cache/nim/koch_d/stdlib_system.nim.c.o /home/euant/.cache/nim/koch_d/stdlib_system.nim.c'
```

This is caused by commit e70294d.

--- a/lib/system/ansi_c.nim
+++ b/lib/system/ansi_c.nim
@@ -117,7 +117,7 @@ type
   CFilePtr* = ptr CFile ## The type representing a file handle.
 
 # duplicated between io and ansi_c
-const stdioUsesMacros = (defined(osx) or defined(bsd)) and not defined(emscripten)
+const stdioUsesMacros = (defined(osx) or defined(freebsd) or defined(dragonfly)) and not defined(emscripten)
 const stderrName = when stdioUsesMacros: "__stderrp" else: "stderr"
 const stdoutName = when stdioUsesMacros: "__stdoutp" else: "stdout"
 const stdinName = when stdioUsesMacros: "__stdinp" else: "stdin"

--- a/lib/system/io.nim
+++ b/lib/system/io.nim
@@ -39,7 +39,7 @@ type
 # text file handling:
 when not defined(nimscript) and not defined(js):
   # duplicated between io and ansi_c
-  const stdioUsesMacros = (defined(osx) or defined(bsd)) and not defined(emscripten)
+  const stdioUsesMacros = (defined(osx) or defined(freebsd) or defined(dragonfly)) and not defined(emscripten)
   const stderrName = when stdioUsesMacros: "__stderrp" else: "stderr"
   const stdoutName = when stdioUsesMacros: "__stdoutp" else: "stdout"
   const stdinName = when stdioUsesMacros: "__stdinp" else: "stdin"

